pkg/controllers/gateway/gateway_controller.go

```diff
@@ -171,8 +171,14 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// gateway now in expected state, place gateway and its associated objects in correct places. Update gateway spec/metadata
 	log.V(3).Info("reconcileDownstreamFromUpstreamGateway result ", "requeue", requeue, "status", programmedStatus, "clusters", clusters, "Err", reconcileErr)
 	if reconcileErr != nil {
+		//TODO (cbrookes) refactor how status is handled in this controller
```

Comment:
Do we have an issue for this?

Reply:
yes although it could prob do with expanding on the details https://github.com/Kuadrant/multicluster-gateway-controller/issues/410